### DISCUSSION TITLE
Problems control of colors and color codes

### DISCUSSION THREAD

**Discussion by Betzebu1959** [alandefreitas/matplotplusplus#335] [Q&A]
I am having troubles in understanding how to control the colors of displayed objects such as pie chart or barstack segments. It appears to me that the assignment of colors to such subsegments of objects (e. g. a cell in a barstacked element or a section of a pie chart) are happening random and cannot be controlled by submitting arguments (when compared with Python’s matplotlib)
Moreover, the codes used by the few examples in the gallery  – e.. g. face_color({0.f, .5f, .5f}) – are totally unclear to me. Is there some kind of howto available that could help especially for the control of colors?

Apr 12:

I've found out a bit more about this in the gallery:I learned that with this code one can control the order of colors in multiplots:

vector_2d newcolors = {{0.83, 0.14, 0.14},
                           {1.00, 0.54, 0.00},
                           {0.47, 0.25, 0.80},
                           {0.25, 0.80, 0.54}};
    colororder(newcolors);

What is driving me nuts, though, is this sequence of 3 floats and what color they stand for. I tried to understand better by experimenting in Gnuplot but nothing helps so far. Any idea where I can obtain a map of float codes to color names in matplot++ ?

**Comment by timmyaveragekid** [comment]
I'm pretty sure these are RGB values.
